packages/grafana-data/src/types/dataLink.ts

```diff
@@ -28,16 +32,19 @@ export interface DataLink {
   // Not saved in JSON/DTO
   onClick?: (event: DataLinkClickEvent) => void;
 
-  // At the moment this is used for derived fields for metadata about internal linking.
-  meta?: {
-    datasourceUid?: string;
+  // If dataLink represents internal link this has to be filled. Internal link is defined as a query in a particular
+  // datas ource that we want to show to the user. Usually this results in a link to explore but can also lead to
+  // more custom onClick behaviour if needed.
+  internal?: {
```

Comment:
Change to the types since last commit. Removed 2 different types and just made the internal link attributes more type safe. At the same time though it's not ideal as you have to check for this and for functions taking DataLink you do not know if they will handle internal links properly.